src/rotation.rs

```diff
@@ -30,53 +30,42 @@ use vector::{Vector2, Vector3};
 
 /// A trait for a generic rotation. A rotation is a transformation that
 /// creates a circular motion, and preserves at least one point in the space.
-pub trait Rotation<P: EuclideanSpace>: Sized + Copy + One
+pub trait Rotation: Sized + Copy + One
 where
     // FIXME: Ugly type signatures - blocked by rust-lang/rust#24092
-    Self: approx::AbsDiffEq<Epsilon = P::Scalar>,
-    Self: approx::RelativeEq<Epsilon = P::Scalar>,
-    Self: approx::UlpsEq<Epsilon = P::Scalar>,
-    P::Scalar: BaseFloat,
+    Self: approx::AbsDiffEq<Epsilon = <<Self as Rotation>::Space as EuclideanSpace>::Scalar>,
```

Comment:
I don't think you need `<Self as Rotation>::` anywhere. Can you just use `Self::`?

Reply:
Nope, the compiler complains:
`error[E0391]: cycle detected when computing the bounds for type parameter 'Self'`

Reply:
ugh